### PR TITLE
statistics: add comments and change `isIndex` to bool

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: None

Problem Summary:

### What is changed and how it works?

- Added more comments.
- Change `isIndex` to bool

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - Only update comment

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
🔢 self-check

**Review by hawkingrei — APPROVED** [review]

File: statistics/handle/globalstats/global_stats.go [hunk 1]
```diff
@@ -54,12 +54,24 @@ type GlobalStats struct {
 	ModifyCount           int64
 }
 
+func newGlobalStats(histIDsLen int) *GlobalStats {
```
Comment (winoros) [inline statistics/handle/globalstats/global_stats.go:57]: I think the var name like `histCount` will be better than this one?

Comment (hi-rustin) [reply statistics/handle/globalstats/global_stats.go]: Good point! Addressed!

**Review by winoros — APPROVED** [review]

**Pull request merged**